src/components/Charts/LineChart.tsx

```diff
@@ -28,80 +39,105 @@ import LineChartOperations from '@/operations/charts/LineChartOperations';
-      </ModalContent>
-    </Modal>
-  );
+  const lineChartOptions: Highcharts.Options = LineChartOperations.getHighChartLineData(lineChartData, theme);
 
-  // Button to trigger the full screen modal; rendered if `expandable==true`
-  const fullScreenButton = expandable ? (
-    <Button isIconOnly variant="light" size="sm" onClick={onOpen}>
-      <Maximize4 className={`h-${ICON_BUTTON_SIZE} w-${ICON_BUTTON_SIZE}`} />
-    </Button>
-  ) : null;
+  // controlling if a line or bar chart is rendered; line chart is the default
+  const [showBarChart, setShowBarChart] = useState(false);
+  const [chartOptions, setChartOptions] = useState(lineChartOptions);
+  // handling the x-axis range slider visibility
+  const [showXAxisSlider, setShowXAxisSlider] = useState(false);
 
-  // Description text element should only be rendered if description is available
-  const descriptionText = description ? (
-    <p className={`w-full h-fit pb-4 ${DESCRIPTION_TEXT_SIZE} px-3`}> {description} </p>
-  ) : null;
+  // handling the line and bar chart switch and the theme switch
+  useEffect(() => {
+    const newChartOptions = showBarChart
+      ? LineChartOperations.getHighChartBarData(lineChartData, theme)
+      : LineChartOperations.getHighChartLineData(lineChartData, theme);
+    setChartOptions(newChartOptions);
+  }, [showBarChart, theme]);
 
   return (
     <>
-      <div className="w-full h-fit flex-col rounded-md bg-background">
-        <div
-          className={`w-full h-fit flex flex-row justify-between items-start gap-3 pl-3 pb-${HEADER_BOTTOM_PADDING}`}
-        >
-          <p className={`${TITLE_TEXT_SIZE} font-normal pt-2 flex flex-row items-center`}> {title} </p>
-          {fullScreenButton}
+      <div className={`w-full h-fit flex-col rounded-md ${transparentBackground ? 'bg-background' : 'bg-transparent'}`}>
+        <div className={`w-full h-fit flex flex-row justify-between items-start gap-1 pl-3 pb-${HEADER_PADDING}`}>
+          <h2 className={`${TITLE_TEXT_SIZE} font-normal pt-2 flex flex-row items-center`}> {title} </h2>
+          <div className="flex flex-row gap-0.5 pt-0.5 pr-0.5">
+            {
+              // button to hide/show the slider to manipulate the plotted x-axis range of the chart;
+              // can be disabled via `xAxisSlider`
+              xAxisSlider ? (
+                <Tooltip text="x-Axis Slider">
+                  <Button
+                    isIconOnly
+                    variant="light"
+                    size="sm"
+                    onPress={() => {
+                      setShowXAxisSlider(!showXAxisSlider);
+                    }}
+                  >
+                    <Settings className={`h-${ICON_BUTTON_SIZE} w-${ICON_BUTTON_SIZE}`} />
+                  </Button>
+                </Tooltip>
+              ) : null
+            }
+            {
+              // button to switch between line and bar chart; can be disabled via `barChartSwitch`
+              barChartSwitch ? (
```

Comment:
use `barChartSwitch && component` instead of ternary operator, it has the same functionality and its cleaner